lib/profiling/template/dummy_lib.f90

```diff
@@ -23,46 +26,48 @@ end subroutine ProfileInit
   !> Starts a profiling area. The module and region name can be used to create
   !> a unique name for each region.
   !> Parameters: 
-  !> module_name:  Name of the module in which the region is
-  !> region_name:  Name of the region (could be name of an invoke, or
-  !>               subroutine name).
-  !> profile_data: Persistent data used by the profiling library.
-  subroutine ProfileStart(module_name, region_name, profile_data)
+  !> this:        PSyData instance.
+  !> module_name: Name of the module in which the region is
+  !> region_name: Name of the region (could be name of an invoke, or
+  !>              subroutine name).
```

Comment:
Done, but I don't like to declare them with a default value: this way these functions look different from the API, and it might even indicate to a user that the variables might not be provided (while they always will be). We could add code that checks that the values are indeed 0 at runtime, but since #415 will kind of handle this differently anyway, I don't see too much sense in this tbh.

Reply:
OK.